packaging/installer/netdata-updater.sh

```diff
@@ -149,6 +149,14 @@ issystemd() {
   return 1
 }
 
+# shellcheck disable=SC2009
+running_under_anacron() {
+    ppid="$(ps -o pid=,ppid= | grep -e "^ $$" | xargs | cut -f 2 -d ' ')"
+    ps -o pid=,command= | grep -e "^ ${ppid}" | grep -q anacron && return 0
```

Comment:
What is this code supposed to check?

---

Bard suggested checking the parent process name (and mention that is is not reliable)

> You can check the name of the parent process using ps or /proc filesystem. Anacron's process name might include "anacron" or "run-parts" (which anacron uses to execute jobs). This is a weak indicator as other processes might have similar names.

```bash
parent_process=$(ps -o comm= $(ps -o ppid= $$ | head -n 1))
if [[ $parent_process == *"anacron"* || $parent_process == *"run-parts"* ]]; then
  # Might be anacron, but not guaranteed
fi
```

Reply:
That’s exactly what this code is checking, because it’s the best we can do, it’s just taking a slightly different approach that is a bit more portable (not all `ps` implementations support taking a PID as a direct argument, and busybox `ps` does not support the POSIX `-p` option that would otherwise make that portable) and ignores `run-parts` (because that also gets used for regular cron runs without anacron on some Debian systems). I should probably have it check two parents up though, which would catch the case of `run-parts` being used.

The first line is determining the parent PID for the script (`xargs` is in there to get rid of extraneous spaces so that `cut` works reliably), the second is checking the command name of that PID.

Reply:
I didn't get the `grep -e "^ $$"` because on my VM the output doesn't start with a space

```bash
$ ps -o pid=,ppid=
3437888 3437887
3440243 3437888
```

So nothing will match with that `grep`.

I checked it on a different node (Proxmox host) and yes it has a space

```bash
$ ps -o pid=,ppid=
 589091  589090
 595459  589091
```

Reply:
Hm, I guess it depends on the pid (small/big). Checked on a Centos7 VM - 2 spaces

```bash
$ ps -o pid=,ppid=
,ppid=
  8386
  846
```

Reply:
The latest commit should fix both issues (it prefers just scanning `/proc` if it’s mounted on linux, allows a variable number of leading spaces when matching `ps` output, and bails if the initial `ps` call doesn’t produce a PPID).